src/ast/passes/codegen_llvm.cpp

```diff
@@ -1088,6 +1088,31 @@ void CodegenLLVM::visit(Call &call)
     b_.CreateLifetimeEnd(buf);
     expr_ = nullptr;
   }
+  else if (call.func == "bswap")
+  {
+    bpftrace::ast::Expression *arg = call.vargs->at(0);
+    auto scoped_del_arg = accept(arg);
+
+    assert(arg->type.IsIntegerTy());
+    if (arg->type.GetSize() > 1)
+    {
+      llvm::Type *arg_type = b_.GetType(arg->type);
+      Function *swap_fun = Intrinsic::getDeclaration(module_.get(),
+                                                     Intrinsic::bswap,
+                                                     { arg_type });
+
+      // Not sure what is going on, but when i16 values were handled, the
+      // correct bswap intrinsic would be selected, (bswap.i16), yet LLVM would
+      // give to it a zero extended i64 argument! The trick below seems to fix
+      // this behaviour and the argument to bswap is always of the correct type.
+      Value *promoted_value = b_.CreateIntCast(expr_, b_.getInt64Ty(), false);
```

Comment:
does this happen in the original IR or the optimized one? We have some weird casts in there for 'reasons' which can make the optimized ir confusing